In [1]:
import pandas as pd 
df = pd.read_csv("Dataset for Data Analytics - Sheet1.csv")

df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [2]:
df.info()
df.shape

df.isnull().sum()

df.duplicated().sum()
df[df.duplicated()]

df['OrderID'].duplicated().sum()
df.dtypes

df.describe(include='all')

<class 'pandas.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   OrderID          1200 non-null   str    
 1   Date             1200 non-null   str    
 2   CustomerID       1200 non-null   str    
 3   Product          1200 non-null   str    
 4   Quantity         1200 non-null   int64  
 5   UnitPrice        1200 non-null   float64
 6   ShippingAddress  1200 non-null   str    
 7   PaymentMethod    1200 non-null   str    
 8   OrderStatus      1200 non-null   str    
 9   TrackingNumber   1200 non-null   str    
 10  ItemsInCart      1200 non-null   int64  
 11  CouponCode       891 non-null    str    
 12  ReferralSource   1200 non-null   str    
 13  TotalPrice       1200 non-null   float64
dtypes: float64(2), int64(2), str(10)
memory usage: 131.4 KB


,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
count,1200,1200,1200,1200,1200.000000,1200.000000,1200,1200,1200,1200,1200.000000,891,1200,1200.000000
unique,1200,671,1189,7,NaN,NaN,655,5,5,1200,NaN,3,5,NaN
top,ORD200000,2023-08-20,C98474,Printer,NaN,NaN,533 Main St,Online,Cancelled,TRK37947903,NaN,FREESHIP,Instagram,NaN
freq,1,8,2,181,NaN,NaN,8,258,250,1,NaN,313,259,NaN
mean,NaN,NaN,NaN,NaN,2.945833,356.412750,NaN,NaN,NaN,NaN,5.485000,NaN,NaN,1053.968300
std,NaN,NaN,NaN,NaN,1.407557,197.177146,NaN,NaN,NaN,NaN,2.281983,NaN,NaN,819.856558
min,NaN,NaN,NaN,NaN,1.000000,11.390000,NaN,NaN,NaN,NaN,1.000000,NaN,NaN,11.390000
25%,NaN,NaN,NaN,NaN,2.000000,186.062500,NaN,NaN,NaN,NaN,4.000000,NaN,NaN,410.520000
50%,NaN,NaN,NaN,NaN,3.000000,364.210000,NaN,NaN,NaN,NaN,5.000000,NaN,NaN,823.615000
75%,NaN,NaN,NaN,NaN,4.000000,521.570000,NaN,NaN,NaN,NaN,7.000000,NaN,NaN,1578.475000


In [3]:
df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

In [4]:
df.duplicated().sum()

np.int64(0)

In [5]:
df['OrderID'].duplicated().sum()

np.int64(0)

In [6]:
df['CouponCode'] = df['CouponCode'].fillna('No Coupon')

In [7]:
df.dtypes

OrderID                str
Date                   str
CustomerID             str
Product                str
Quantity             int64
UnitPrice          float64
ShippingAddress        str
PaymentMethod          str
OrderStatus            str
TrackingNumber         str
ItemsInCart          int64
CouponCode             str
ReferralSource         str
TotalPrice         float64
dtype: object

In [10]:
df['Date'] = pd.to_datetime(df['Date'], errors='coerce')
df['Date'].isnull().sum()
df[['Quantity', 'UnitPrice', 'TotalPrice']].dtypes

Quantity        int64
UnitPrice     float64
TotalPrice    float64
dtype: object

In [9]:
df['PaymentMethod'].unique()

<StringArray>
['Debit Card', 'Online', 'Credit Card', 'Gift Card', 'Cash']
Length: 5, dtype: str

In [11]:
df['OrderStatus'].unique()
df['Product'].unique()

<StringArray>
['Monitor', 'Phone', 'Tablet', 'Chair', 'Printer', 'Laptop', 'Desk']
Length: 7, dtype: str

In [12]:
df['OrderStatus'].unique()

<StringArray>
['Shipped', 'Cancelled', 'Returned', 'Delivered', 'Pending']
Length: 5, dtype: str

In [15]:
(df['Quantity'] *  df['UnitPrice'] - df['TotalPrice']). abs().max()

np.float64(4.547473508864641e-13)

## Data Cleaning Log - Orders Dataset

1. **Initial inspection**
- Ran df.head(), df.info() - no structural issues
- Ran df.isnull().sum() - found 309 missing values in CouponCOde only

2. **Duplicate check**
- df.duplicated() .sum() > 0 (no fully duplicated rows)
- df ['OrderId'].duplicated().sum() > 0 (no duplicate order IDs)

3. **Missing values**
- Filled 309 missing CouponCode values  qith "No Coupon"
- Verified: df['CouponCode'].isnull().sum() > 0

4. **Format correction**
- Converted Date column from string to datetime using pd.to_datetime()
- Verified: 0 dates failed to parse

5. **Text consistency check**
- Checked unique values in PaymentMethod , Product, Orderstatus
- All three columns comfirmed consistent  - no casing/spacing issues, no changes needed

6.**Numeric validation**
- Verified Quantity x  unitprice ~~ TotalPrice across all rows
- Max discrepancy: ~4.5e-13 (floating-point rounding,not a data error)

**Results:** Dataset passes verification gate - 0% duplicate IDs, 0% bad date formats.